src/daemon/Seat.cpp

```diff
@@ -47,6 +47,23 @@ namespace SDDM {
     }
 
     void Seat::createDisplay(Display::DisplayServerType serverType) {
+        if (Logind::isAvailable()) {
```

Comment:
Should IMO be done internally in `SeatManager::switchToGreeter`, iterate all displays and check whether it's a greeter.